ocean_lib/common/ddo/credentials.py

```diff
@@ -0,0 +1,106 @@
+#
+# Copyright 2021 Ocean Protocol Foundation
+# SPDX-License-Identifier: Apache-2.0
+#
+from ocean_lib.common.agreements.consumable import ConsumableCodes, MalformedCredential
+
+
+class AddressCredential:
+    def __init__(self, asset):
+        self.asset = asset
+
+    def get_addresses_of_class(self, access_class="allow"):
+        """Get a filtered list of addresses from credentials (use with allow/deny)."""
+        entries = self.asset._credentials.get(access_class, [])
+
+        for entry in entries:
+            if entry["type"] == "address":
+                return [val.lower() for val in entry["values"]]
+
+        return []
+
+    def requires_credential(self):
+        allowed_addresses = self.get_addresses_of_class("allow")
+        denied_addresses = self.get_addresses_of_class("deny")
+
+        return allowed_addresses or denied_addresses
+
+    def validate_access(self, credential=None):
+        address = simplify_credential_to_address(credential)
+
+        allowed_addresses = self.get_addresses_of_class("allow")
+        denied_addresses = self.get_addresses_of_class("deny")
+
+        if not address and not self.requires_credential():
+            return ConsumableCodes.OK
+
+        if allowed_addresses and address.lower() not in allowed_addresses:
+            return ConsumableCodes.CREDENTIAL_NOT_IN_ALLOW_LIST
+
+        if not allowed_addresses and address.lower() in denied_addresses:
+            return ConsumableCodes.CREDENTIAL_IN_DENY_LIST
+
+        return ConsumableCodes.OK
+
+    def add_address_to_list_class(self, address, list_class="allow"):
```

Comment:
Consider standardizing all uses of "list_class" to "access_class" because it's more descriptive.